examples/beam/src/main/java/org/apache/nemo/examples/beam/WindowedWordCount.java

```diff
@@ -43,36 +45,45 @@ private WindowedWordCount() {
   public static final String INPUT_TYPE_BOUNDED = "bounded";
   public static final String INPUT_TYPE_UNBOUNDED = "unbounded";
   private static final String SPLITTER = "!";
-
+  public static final Random RAND = new Random();
 
   /**
    * @param p    pipeline.
-   * @param args arguments.
+   * @param inputType input type arg.
+   * @param inputFilePath input file path arg.
    * @return source.
    */
   private static PCollection<KV<String, Long>> getSource(
     final Pipeline p,
-    final String[] args) {
+    final String inputType,
+    final String inputFilePath) {
 
-    final String inputType = args[2];
     if (inputType.compareTo(INPUT_TYPE_BOUNDED) == 0) {
-      final String inputFilePath = args[3];
       return GenericSourceSink.read(p, inputFilePath)
         .apply(ParDo.of(new DoFn<String, String>() {
           @ProcessElement
           public void processElement(@Element final String elem,
                                      final OutputReceiver<String> out) {
             final String[] splitt = elem.split(SPLITTER);
-            out.outputWithTimestamp(splitt[0], new Instant(Long.valueOf(splitt[1])));
+            if (splitt.length > 1 && splitt[1].matches("[0-9]+")) {
+              out.outputWithTimestamp(splitt[0], new Instant(Long.valueOf(splitt[1])));
+            } else {
+              final long timestamp = System.currentTimeMillis() - RAND.nextInt(1000000);
+              out.outputWithTimestamp(elem, new Instant(timestamp));
```

Comment:
Is it okay for the correctness of end-to-end operation if we assign dummy timestamp for documents with format error in the `date` field?

Reply:
This code produces timestamps within 1000s of the past in order to demonstrate out-of-order data and so. I've added this as a comment.